app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,66 @@
+class MarketsController < ApplicationController
+  def home
+    @markets = Market.order(:name)
+  end
+
+  def index
+  	user = params[:user_type]
+  	session[:user_type] = user unless user.nil?
+  	@buttons = true if session[:user_type] == 'market'
+  	@markets = Market.all if params[:order].nil?
+    @markets = Market.order(name: :desc) if params[:order] == 'za'
+    @markets = Market.order(:name) if params[:order] == 'az'
+  end
+
+  def new
+    @market = Market.new
+  end
+
+  def show
+  	@market = Market.find(params[:id])
+		@is_market = true if session[:user_type] == 'market'
```

Comment:
Same comment re: combined conditionals. Using if/else/elsif is much more efficient and does not cause unnecessary conditional execution.